### PR TITLE
Add Open AI integration

### PR DESCRIPTION
**Pull request by jozefKruszynski**
First draft of the open ai integration
Would love to get some feedback, as I'm sure there are many things that I have done in a sloppy manner.
It's not really ready for merging, but I definitely need feedback on how to improve it.

### CODE REVIEW

File: custom_components/mass/intent.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+"""Intents for the client integration."""
```
Comment (marcelveldt) [inline custom_components/mass/intent.py:1]:
```suggestion
"""Intents for the Music Assistant integration."""
```

File: custom_components/mass/intent.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        name: str = slots.get("name", {}).get("value")
+        if name is not None:
+            media_id, media_type = await self.get_media_id_and_media_type_from_query_result(
+                hass, service_data, intent_obj
+            )
+            mass: MusicAssistantClient = hass.data[DOMAIN][config_entry.entry_id].mass
+            players = await mass.players.get_players()
+            for player in players:
+                if player.name.casefold() == name.casefold():
+                    actual_player = MassPlayer(mass, player.player_id)
```
Comment (marcelveldt) [inline custom_components/mass/intent.py:59]: what happens if the actual player is not set?
maybe you should break when the player is found/set and add an "else" to the for loop

**Comment by marcelveldt** [comment]
Please hit the "ready for review" button once you;re ready for the final review to get tis merged @jozefKruszynski

File: custom_components/mass/intent.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+QUERY_SLOT = "query"
+SLOT_VALUE = "value"
+
+
+async def async_setup_intents(hass: HomeAssistant) -> None:
+    """Set up the climate intents."""
+    intent.async_register(hass, MassPlayMediaOnMediaPlayerEn())
+
+
+class MassPlayMediaOnMediaPlayerEn(intent.IntentHandler):
```
Comment (marcelveldt) [inline custom_components/mass/intent.py:41]: Why is this class named "MassPlayMediaOnMediaPlayerEn" and not "MassPlayMediaOnMediaPlayerHandler" or something ? In other words: what does the "En" postfix mean in this case ? It is limited to English ?

File: custom_components/mass/intent.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            return False
+
+        return any(name == alias.casefold() for alias in entity.aliases)
+
+    async def _get_mass_player_from_registry_entry(
+        self, hass: HomeAssistant, config_entry: ConfigEntry, media_player_entity: er.RegistryEntry
+    ) -> MassPlayer:
+        """Return the mass player."""
+        mass: MusicAssistantClient = hass.data[DOMAIN][config_entry.entry_id].mass
+        mass_player = mass.players.get_player(media_player_entity.unique_id.strip("mass_"))
```
Comment (marcelveldt) [inline custom_components/mass/intent.py:161]: maybe better just grab the "mass_player_id" attribute of the HA entity ?

Comment (jozefKruszynski) [reply custom_components/mass/intent.py:161]: The problem is that here we only have the RegistryEntry which does not expose the actual entity attributes.

File: custom_components/mass/intent.py (hunk 3 above)
Comment (jozefKruszynski) [reply custom_components/mass/intent.py:41]: Remnant from when I started creating the class. Changing it to your suggestion

File: custom_components/mass/intent.py (hunk 4 above)
Comment (jozefKruszynski) [reply custom_components/mass/intent.py:161]: I'll look into if I can do this smarter

Comment (jozefKruszynski) [reply custom_components/mass/intent.py]: Found a better way :)

**Review by marcelveldt — APPROVED** [review]
Nice work @jozefKruszynski !

**Pull request merged**